Initial structure of README with some content examples

amivanoff opened the pull request:
README with the Contents and initial structure of sections.

Some sections entries migrated from [Updated list of implementations](https://github.com/validatingrdf/validatingrdf.github.io/wiki/Updated-list-of-implementations) as an examples. It is not a complete migration, just an illustration.

Closes #1

tpluscode commented on README.md:
```diff
@@ -1,5 +1,93 @@
+
+## SHACL Validators
+
+About this section. Optional. Keep this short and focus on the list.
```
To be removed?

amivanoff replied:
Yes, my bad. I'll fix it.

tpluscode commented on README.md:
```diff
@@ -1,5 +1,93 @@
+
+## Contents
+
+- [SHACL Validators](#shacl-validators)
+- [ShEx Validators](#shex-validators)
```
Do we need two validator categories?

amivanoff replied:
It is useful from "validator user" point of view to separate it. I assume (maybe wrongly), an average project had committed to SHACL or ShEx from the beginning. And project's CTO "selecting from" or "monitoring for a tool switch/upgrade" a particular category.

tpluscode replied:
Okay, I realised my mistake. You are right, of course. Makes sense

amivanoff replied:
Also, it seems, it will be to long list if we combine it together.

tpluscode reviewed (approved):
LGTM

The Contribution Guidelines is not there yet and mabe we can piggy back on [the awesome manifesto](https://github.com/sindresorhus/awesome/blob/main/awesome.md)?

VladimirAlexiev commented on README.md:
```diff
@@ -1,5 +1,97 @@
+
+## SHACL Validators
+
+- [Jena SHACL](https://github.com/apache/jena/) -- Java, Supports: SHACL Core, SHACL-SPARQL
```
We want the link to be most specific, into the SHACL documentation of the concrete product

amivanoff replied:
Do I understand correctly that all "validator links" in these two sections should point to the (list in a descending priority):

1. SHACL/ShEx documentation of this validator (for Jena or RDF4J, or Corese), and no need to add link to the sources repository in a source control system?
2. If no validator documentation was found, then the link should point to the validator module folder in the sources repository in a source control system?

amivanoff replied:
> * Afaik, English grammar uses `: ` not `--`

I checked with the [awesome-lint](https://github.com/sindresorhus/awesome-lint), it should be one dash `-` between a link and a description, not em-dash, not double-dash '--'. And it should be a comma in the end if there is a description (or no comma if a link only).

This simple construction -- is the most often one

[software] - Description

Or it could be a more complex construction (for example, here [brillout/awesome-react-components](https://github.com/brillout/awesome-react-components))

[software] - [docs] - [demo] - Description

TallTed replied:
Using a colon (without preceding space, with following space) instead of the em-dash is valid. There should probably be a period (not a comma) following the language in which the thing was written. There's no need for the colon following "Supports". There should probably be a period at the end of the line.
```
- [Jena SHACL](https://github.com/apache/jena/): Java. Supports SHACL Core, SHACL-SPARQL.
```

The "more complex construction" is yet more ugliness, to my eyes. I would much prefer a proper table, where any cell's text could wrap within the cell. The presented format is useful where tables are hard to write and/or render — but that's not the case for any of this content.

I don't know why [awesome-lint](https://github.com/sindresorhus/awesome-lint) should be treated as an authority on human-facing punctuation. I'm sure it does a pretty good job on linting *code*, but Markdown is, by design, closer to human language than to programming. I know coders tend to prefer hyphens, because they don't require a key-chord or more complex entry method, but hyphens aren't appropriate for the placements described.

If the reason the not-so-[awesome-lint](https://github.com/sindresorhus/awesome-lint) is being granted such authority is that this is an "awesome"-list, well, I think the "awe" it will trigger will be less than the "eww", and I suggest reconsideration and pushing back upstream.

amivanoff replied:
[awesome-lint](https://github.com/sindresorhus/awesome-lint) is a specialized markdown linter, tailored specifically to lint "awesome-XXX" lists of some "awesome stuff", created by the author of "awesome list of awesome lists". Some ad-hoc domain-specific convention.

VladimirAlexiev commented on README.md:
```diff
@@ -1,5 +1,97 @@
 
-This is the repository for awesome-semantic-shapes. You're welcome to contribute! Let's make the Web rock our socks
-off!
+Semantic shapes enables you to validate RDF graphs against a set of conditions. Semantic shapes can also be viewed as a description of the data graphs that do satisfy these conditions. Such descriptions may be used for a variety of purposes beside validation, including user interface building, code generation and data integration. Semantic shapes could be described in SHACL or ShEx languages.
```
- "enable"
- use Semantic Newlines: https://sembr.org/

TallTed replied:
Ugh. Semantic Newlines documents "the `git diﬀ` command" (note the ligature) which doesn't exist; it's the `git diff` command.

I do encourage more frequent line breaks, so lines are shorter and generally fit into a single window width. I find 80 chars or so is generally best for most devices. I find semantic newlines *less* readable, because they frequently deliver wider lines.

amivanoff replied:
Sorry for my stupidity, but where did you get `ﬀ`? My browser shows two separate `ff` in all commands on the https://sembr.org webpage, on the github page, in the raw README.md markdown.

TallTed replied:
No stupidity in your question. Possibly some in programmers of Chrome or one of my active Chrome plugins, that's displaying ligatures without my express permission.

<img width="889" alt="image" src="https://github.com/user-attachments/assets/7c1b170e-8c1e-4ef4-8e7a-b837de9d28df">

amivanoff replied:
In mine it looks normal and I could select separate 'f' characters.
![image](https://github.com/user-attachments/assets/efb08fd0-f5d1-4843-b2cb-b8e48d6120b6)

TallTed commented on README.md:
```diff
@@ -1,8 +1,13 @@
 
-Semantic shapes enables you to validate RDF graphs against a set of conditions. Semantic shapes can also be viewed as a description of the data graphs that do satisfy these conditions. Such descriptions may be used for a variety of purposes beside validation, including user interface building, code generation and data integration. Semantic shapes could be described in SHACL or ShEx languages.
+Semantic shapes enable you to validate RDF graphs against a set of conditions.
+Semantic shapes can also be viewed as a description of the data graphs that do satisfy these conditions.
```
```suggestion
Semantic shapes can also be viewed as a description of data graphs that satisfy these conditions.
```

TallTed commented on README.md:
```diff
@@ -1,8 +1,13 @@
-Semantic shapes enables you to validate RDF graphs against a set of conditions. Semantic shapes can also be viewed as a description of the data graphs that do satisfy these conditions. Such descriptions may be used for a variety of purposes beside validation, including user interface building, code generation and data integration. Semantic shapes could be described in SHACL or ShEx languages.
+Semantic shapes enable you to validate RDF graphs against a set of conditions.
+Semantic shapes can also be viewed as a description of the data graphs that do satisfy these conditions.
+Such descriptions may be used for a variety of purposes beside validation,
+including user interface building, code generation and data integration.
+Semantic shapes could be described in SHACL or ShEx languages.
```
```suggestion
Such descriptions may be used for a variety of purposes besides validation,
including user interface building, code generation, and data integration.
Semantic shapes are frequently described using the SHACL or ShEx language.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
-- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) -- Java
-- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) -- Java, based on Jena, supports: SHACL Core, SHACL-SPARQL, SHACL rules
-- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) -- Java, fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot
+- [Jena SHACL](https://github.com/apache/jena/) - Java, Supports: SHACL Core, SHACL-SPARQL.
+- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) - Java.
+- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) - Java, based on Jena, supports: SHACL Core, SHACL-SPARQL, SHACL rules.
+- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) - Java, fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot.
```
```suggestion
- [Jena SHACL](https://github.com/apache/jena/) — Java; Supports: SHACL Core, SHACL-SPARQL.
- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) — Java.
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) — Java; based on Jena; supports: SHACL Core, SHACL-SPARQL, SHACL rules.
- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) — Java; fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
-- [shexSpec/shex.js](https://github.com/shexjs/shex.js) -- JS, [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html)
-- [ShEx-validator](https://github.com/HW-SWeL/ShEx-validator) -- JS
-- [Validata](https://github.com/HW-SWeL/Validata) -- JS, [playground](http://hw-swel.github.io/Validata/), [playground](https://www.w3.org/2015/03/ShExValidata/)
+- [Apache Jena ShEx](https://jena.apache.org/documentation/shex/index.html) - Java.
+- [shexSpec/shex.js](https://github.com/shexjs/shex.js) - [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html) - JS.
+- [ShEx-validator](https://github.com/HW-SWeL/ShEx-validator) - JS.
+- [Validata](https://github.com/HW-SWeL/Validata) - [playground](http://hw-swel.github.io/Validata/) - [playground](https://www.w3.org/2015/03/ShExValidata/), JS.
```
```suggestion
- [Apache Jena ShEx](https://jena.apache.org/documentation/shex/index.html) — Java.
- [shexSpec/shex.js](https://github.com/shexjs/shex.js) — [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html) — JS.
- [ShEx-validator](https://github.com/HW-SWeL/ShEx-validator) — JS.
- [Validata](https://github.com/HW-SWeL/Validata) — [playground](http://hw-swel.github.io/Validata/) — [playground](https://www.w3.org/2015/03/ShExValidata/), JS.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
 ## Shapes Discovery Tools
 
-- [RDFminer](https://github.com/Wimmics/RDFminer) -- Web application to automatically discovering SHACL shapes representative of an RDF data graph, by Wimmics
+- [RDFminer](https://github.com/Wimmics/RDFminer) - Web application to automatically discovering SHACL shapes representative of an RDF data graph, by Wimmics.
```
```suggestion
- [RDFminer](https://github.com/Wimmics/RDFminer) — Web application to automatically discover SHACL shapes representative of an RDF data graph, by Wimmics.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
+- [RDFminer](https://github.com/Wimmics/RDFminer) - Web application to automatically discovering SHACL shapes representative of an RDF data graph, by Wimmics.
 - [SHACL Discovery Service](https://github.com/AKSW/discover-shacl-shapes)
-- [Shapes of You index](https://index.semanticscience.org/) -- SPARQL queries, OWL/SKOS vocabularies, SHACL/ShEx shapes indexed from public git repositories.
+- [Shapes of You index](https://index.semanticscience.org/) - SPARQL queries, OWL/SKOS vocabularies, SHACL/ShEx shapes indexed from public git repositories.
```
```suggestion
- [Shapes of You index](https://index.semanticscience.org/) — SPARQL queries, OWL/SKOS vocabularies, SHACL/ShEx shapes, indexed from public `git` repositories.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
 ## Shapes Collections
 
-- [schema.org Shapes](http://datashapes.org/schema) -- Schema.org, converted to SHACL by TopQuadrant
+- [schema.org Shapes](http://datashapes.org/schema) - Schema.org, converted to SHACL by TopQuadrant.
```
```suggestion
- [schema.org Shapes](http://datashapes.org/schema) — Schema.org, converted to SHACL by TopQuadrant.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
 
-- [ShacShifter](https://github.com/AKSW/ShacShifter) -- Python, ☠️, "shape shifter" from SHACL to other formats (currently RDForms)
-- [SHACL To JSON Schema](https://github.com/comake/shacl-to-json-schema) -- TS, SHACL To JSON Schema translator
+- [ShacShifter](https://github.com/AKSW/ShacShifter) - Python, ☠️, "shape shifter" from SHACL to other formats (currently RDForms).
+- [SHACL To JSON Schema](https://github.com/comake/shacl-to-json-schema) - TS, SHACL To JSON Schema translator.
```
```suggestion
- [ShacShifter](https://github.com/AKSW/ShacShifter) — Python, ☠️, "shape shifter" from SHACL to other formats (currently RDForms).
- [SHACL To JSON Schema](https://github.com/comake/shacl-to-json-schema) — TS, SHACL-to-JSON-Schema translator.
```

TallTed commented on README.md:
```diff
@@ -22,36 +28,36 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
 ## Shape Generators
 
-- [owl2shacl](https://github.com/sparna-git/owl2shacl) -- OWL 2 SHACL conversion rules
+- [owl2shacl](https://github.com/sparna-git/owl2shacl) - OWL 2 SHACL conversion rules.
```
```suggestion
- [owl2shacl](https://github.com/sparna-git/owl2shacl) — OWL-to-SHACL conversion rules.
```

TallTed commented on README.md:
```diff
@@ -61,11 +67,15 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
 Data viewers/Editors based on shapes.
 
-- [shaperone](https://forms.hypermedia.app) -- SHACL Shapes Form generator
+- [shaperone](https://forms.hypermedia.app) - SHACL Shapes Form generator.
```
```suggestion
- [shaperone](https://forms.hypermedia.app) — SHACL Shapes Form generator.
```

TallTed commented on README.md:
```diff
@@ -61,11 +67,15 @@ Semantic shapes enables you to validate RDF graphs against a set of conditions.
+
+## IDE support
+
+- [Linked Data Extension](https://github.com/elsevierlabs-os/linked-data) - VS Code Extension for RDF files editing with embedded SHACL validator and SPARQL engine.
```
```suggestion
- [Linked Data Extension](https://github.com/elsevierlabs-os/linked-data) — VS Code Extension for editing RDF files with embedded SHACL validator and SPARQL engine.
```

TallTed commented on README.md:
```diff
@@ -76,22 +86,25 @@ Data viewers/Editors based on shapes.
-  - [SHACL JavaScript Extensions](https://www.w3.org/TR/shacl-js/) -- W3C Working Group Note, 08 June 2017
-  - [SHACL Test Suite and Implementation Report](https://w3c.github.io/data-shapes/data-shapes-test-suite/) -- W3C Document 17 January 2024
-  - [SHACL Use Cases and Requirements](https://www.w3.org/TR/shacl-ucr/) -- W3C Working Group Note 20 July 2017
+  - [Shapes Constraint Language (SHACL)](https://www.w3.org/TR/shacl/) - W3C Recommendation, 20 July 2017.
+  - [SHACL Advanced Features](https://www.w3.org/TR/shacl-af/) - W3C Working Group Note, 08 June 2017.
+  - [SHACL JavaScript Extensions](https://www.w3.org/TR/shacl-js/) - W3C Working Group Note, 08 June 2017.
+  - [SHACL Test Suite and Implementation Report](https://w3c.github.io/data-shapes/data-shapes-test-suite/) - W3C Document 17 January 2024.
+  - [SHACL Use Cases and Requirements](https://www.w3.org/TR/shacl-ucr/) - W3C Working Group Note 20 July 2017.
```
```suggestion
  - [Shapes Constraint Language (SHACL)](https://www.w3.org/TR/shacl/) — W3C Recommendation, 20 July 2017.
  - [SHACL Advanced Features](https://www.w3.org/TR/shacl-af/) — W3C Working Group Note, 08 June 2017.
  - [SHACL JavaScript Extensions](https://www.w3.org/TR/shacl-js/) — W3C Working Group Note, 08 June 2017.
  - [SHACL Test Suite and Implementation Report](https://w3c.github.io/data-shapes/data-shapes-test-suite/) — W3C Document 17 January 2024.
  - [SHACL Use Cases and Requirements](https://www.w3.org/TR/shacl-ucr/) — W3C Working Group Note 20 July 2017.
```

TallTed commented on README.md:
```diff
@@ -76,22 +86,25 @@ Data viewers/Editors based on shapes.
+
 - ShEx
-  - [Shape Expressions Language 2.1](https://shex.io/shex-semantics/index.html) -- Final Community Group Report 8 October 2019
+  - [Shape Expressions Language 2.1](https://shex.io/shex-semantics/index.html) - Final Community Group Report 8 October 2019.
```
```suggestion
  - [Shape Expressions Language 2.1](https://shex.io/shex-semantics/index.html) — Final Community Group Report 8 October 2019.
```

TallTed commented on README.md:
```diff
@@ -76,22 +86,25 @@ Data viewers/Editors based on shapes.
 ## Legend
-☠️ -- each one denotes every 5 year from last update
-💰 -- denotes closed source commercial software
+☠️ -- Each one denotes every 5 year from last update.
+💰 -- Denotes closed source commercial software.
```
```suggestion
☠️ — Each one denotes 5 years (e.g., ☠️☠️☠️ denotes 15 years) since last update.
💰 — Denotes closed source commercial software.
```

amivanoff commented:
What development process is best suitable here?
- should I resolve worked out comments by myself
- or the comment's author will decide and reslove (or not) his comments when he'll check commited corrections

amivanoff commented:
I added a formatting example in the "SHACL Validators" section

- [Jena SHACL](https://github.com/apache/jena/) ![GitHub last commit](https://img.shields.io/github/last-commit/apache/jena) `Apache-2.0` `Java` - Supports: SHACL Core, SHACL-SPARQL.
- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) ![GitHub last commit](https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j) `BSD-3-Clause` `Java`
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) ![GitHub last commit](https://img.shields.io/github/last-commit/TopQuadrant/shacl) `BSD-3-Clause` `Java` - Based on Jena; supports: SHACL Core, SHACL-SPARQL, SHACL rules.
- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) ![GitHub last commit](https://img.shields.io/github/last-commit/SHACL-X/shacl-x) `BSD-3-Clause` `Java` - Fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot.
- [SHACL for Ruby](https://github.com/ruby-rdf/shacl/) ![GitHub last commit](https://img.shields.io/github/last-commit/ruby-rdf/shacl) `BSD-3-Clause` `Ruby` - A pure-Ruby library for working with the Shape Constraint Language to validate the shape of RDF graphs.

Or maybe this one is better?

- [Jena SHACL](https://github.com/apache/jena/) ![GitHub last commit](https://img.shields.io/github/last-commit/apache/jena) - Supports: SHACL Core, SHACL-SPARQL. `Apache-2.0` `Java`
- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) ![GitHub last commit](https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j) - `BSD-3-Clause` `Java`
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) ![GitHub last commit](https://img.shields.io/github/last-commit/TopQuadrant/shacl) - Based on Jena; supports: SHACL Core, SHACL-SPARQL, SHACL rules. `BSD-3-Clause` `Java`
- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) ![GitHub last commit](https://img.shields.io/github/last-commit/SHACL-X/shacl-x) - Fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot. `BSD-3-Clause` `Java`
- [SHACL for Ruby](https://github.com/ruby-rdf/shacl/) ![GitHub last commit](https://img.shields.io/github/last-commit/ruby-rdf/shacl) - A pure-Ruby library for working with the Shape Constraint Language to validate the shape of RDF graphs. `BSD-3-Clause` `Ruby`

Unfortunately, all bages in general are out of line with the text 😞 I could not find any way to align it vertically.

tpluscode commented:
Let's not obsess on details. We can improve gradually

TallTed commented:
> But em-dashes don't pass the [awesome-lint](https://github.com/sindresorhus/awesome-lint) linter. So I propose we should go with ordinary dashes.

em-dashes are the correct punctuation where I've put them, where I replaced hyphens, which are not dashes at all.

Previously, there were double-hyphens at these locations which are the ancient typist's stand-in for an em-dash on typewriters which didn't have the extra strikers for en-dash (~1.5 hyphens wide) and em-dash (~2 hyphens wide). Clever software replaces such doubled-hyphens with em-dashes en-route to the viewer. (Some extra-clever software replaces double-hyphens with en-dash, and triple-hyphens with em-dash.) Less clever software requires that we humans do the job.

I'd like to see the error reported by the [not-so-awesome linter](https://github.com/sindresorhus/awesome-lint), so I can raise an appropriate issue against it.

TallTed commented:
> Unfortunately, all [badges] in general are out of line with the text

Well, *that's* not awesome!

amivanoff commented:
@TallTed, I completely agree that em-dash is way better for the readability and overall.

But there are two problems (numbered below):

The linter error is:
```
> awesome-lint                                                                                                                                                                                                                         
✖ Linting
  README.md:1:1
  ✖  32:47  List item link and description must be separated with a dash  remark-lint:awesome-list-item
```

1. There is even a special check in linter's code [List item link and description separated by invalid en-dash or em-dash](https://github.com/sindresorhus/awesome-lint/blob/d83d0361be944593b27d3060942a6a7dce33216d/rules/list-item.js#L194), which is not throwing this type of error messages in an output stream for some unknown reason.

2. We should also take into consideration that we are planning to submit this list to the community-made aggregators from the issue
    - #6.

So, we could end up with attempts to improve all these aggregators.

What do you think, @VladimirAlexiev, colleagues? What we should do, the right thing or the easy one?

amivanoff commented:
The best what could be done with the activity bage vertical alignment (as-before and as-improved)
- [Jena SHACL](https://github.com/apache/jena/) ![GitHub last commit](https://img.shields.io/github/last-commit/apache/jena) - Supports SHACL Core, SHACL-SPARQL `Apache-2.0` `Java`.
- [Jena SHACL](https://github.com/apache/jena/) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/apache/jena" align="top"> - Supports: SHACL Core, SHACL-SPARQL `Apache-2.0` `Java`.

Is it good enough?

amivanoff commented:
And a bigger list will look like this (the page will be wider)
- [Jena SHACL](https://github.com/apache/jena/) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/apache/jena" align="top"> - Supports SHACL Core, SHACL-SPARQL. `Apache-2.0` `Java`.
- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j" align="top"> - `BSD-3-Clause` `Java`.
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/TopQuadrant/shacl" align="top"> - Based on Jena; supports SHACL Core, SHACL-SPARQL, SHACL rules `BSD-3-Clause` `Java`.
- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/SHACL-X/shacl-x" align="top"> - Fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot `BSD-3-Clause` `Java`.
- [SHACL for Ruby](https://github.com/ruby-rdf/shacl/) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/ruby-rdf/shacl" align="top"> - A pure-Ruby library for working with the Shape Constraint Language to validate the shape of RDF graphs `BSD-3-Clause` `Ruby`.

amivanoff commented:
We have two open questions related to this PR:

#### 1. List entry formatting
Is this last variant (single dashes, bages alignment, semicolons) acceptable or not as a starter? Should we dump bages or not?

#### 2. The entry content question
Rised by @VladimirAlexiev: "[We want the link to be most specific, into the SHACL documentation of the concrete product](https://github.com/w3c-cg/awesome-semantic-shapes/pull/3#discussion_r1771824953)".

Should we make a link to the library's documentation a priority compared to a library's repository link? If so, I could replace repo-links with doc-links whenever it's possible.

Current variant (shaperone from "Declarative UIs" category)
- [Jena SHACL](https://github.com/apache/jena/) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/apache/jena" align="top"> - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` `Java`.
- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j" align="top"> - `BSD-3-Clause` `Java`.
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/TopQuadrant/shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL, SHACL rules; based on Jena; `BSD-3-Clause` `Java`.
- [rdf-validate-shacl](https://github.com/zazuko/rdf-validate-shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/zazuko/rdf-validate-shacl" align="top"> - Supports SHACL Core; pure JavaScript validator on top of the [RDFJS](https://rdf.js.org/) stack; [playground](https://zazuko.github.io/shacl-playground); `MIT` `JavaScript`.
- [shacl-engine](https://github.com/rdf-ext/shacl-engine) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/rdf-ext/shacl-engine" align="top"> - Supports SHACL Core, SHACL-SPARQL; A fast engine for data provided as [RDF/JS](http://rdf.js.org/data-model-spec/) objects; `MIT` `JavaScript`.
- [shaperone](https://github.com/hypermedia-app/shaperone) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/hypermedia-app/shaperone" align="top"> - SHACL Shapes Form generator; [playground](https://forms.hypermedia.app/playground); `MIT` `Typescript`.

Doc-links variant (Jena, RDF4J, shaperone changed, others do not have any documentation)
- [Jena SHACL](https://jena.apache.org/documentation/shacl/index.html) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/apache/jena" align="top"> - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` `Java`.
- [RDF4J SHACL Sail](https://rdf4j.org/documentation/programming/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j" align="top"> - `BSD-3-Clause` `Java`.
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/TopQuadrant/shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL, SHACL rules; based on Jena; `BSD-3-Clause` `Java`.
- [rdf-validate-shacl](https://github.com/zazuko/rdf-validate-shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/zazuko/rdf-validate-shacl" align="top"> - Supports SHACL Core; pure JavaScript validator on top of the [RDFJS](https://rdf.js.org/) stack; [playground](https://zazuko.github.io/shacl-playground); `MIT` `JavaScript`.
- [shacl-engine](https://github.com/rdf-ext/shacl-engine) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/rdf-ext/shacl-engine" align="top"> - Supports SHACL Core, SHACL-SPARQL; A fast engine for data provided as [RDF/JS](http://rdf.js.org/data-model-spec/) objects; `MIT` `JavaScript`.
- [shaperone](https://forms.hypermedia.app) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/hypermedia-app/shaperone" align="top"> - SHACL Shapes Form generator; [playground](https://forms.hypermedia.app/playground); `MIT` `Typescript`.

amivanoff commented:
I have finalized the last version. It will be rendered like this:
https://github.com/w3c-cg/awesome-semantic-shapes/blob/89e8518336025ec1289fdb81141014aa672aedd4/README.md

VladimirAlexiev commented:
@amivanoff thanks, you've done a great job!
I think the first link should be to source because the "last commit" shield suggests that, and people will expect it.
And we can add a link "docs" in the description.

I'm ready to merge, but since you've collected both links, could you make this last change?
Thanks!!

TallTed commented:
> [Apache Jena SHACL](https://jena.apache.org/documentation/shacl/index.html) GitHub last commit - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` `Java`.

Some punctuation is needed between the license and the language. The single space separator is not sufficient. I suggest the word "license" and a semicolon —

> [Apache Jena SHACL](https://jena.apache.org/documentation/shacl/index.html) GitHub last commit - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` license; `Java`.

— or the word "license" and a "full stop" —

> [Apache Jena SHACL](https://jena.apache.org/documentation/shacl/index.html) GitHub last commit - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` license. `Java`.

amivanoff commented:
> I think the first link should be to the source because the "last commit" shield suggests that, and people will expect it.

Awesome Linter checks that all list entries have unique links. Complex software like Jena, implements SHACL and ShEx and appears twice in the SHACL Validators and ShEx Validators sections.

With doc-links approach it was OK. With repo-links we have a problem.

We could use links to the corresponding submodules, but they are not so informative for the reader:
- https://github.com/apache/jena/tree/main/jena-shacl
- https://github.com/apache/jena/tree/main/jena-shex

We could use the same "parent" link https://github.com/apache/jena in the SHACL Validators and the ShEx Validators sections but the linter returns an error.

Another option, we could use doc-links and clickable badges with URL links to the repositories.

amivanoff commented:
@TallTed, I will add "license" and a semicolon to list entries.

VladimirAlexiev commented:
Does the validation work with links like 
:https://github.com/apache/jena/tree/main/jena-shacl ? If so, let's go for that. 
The third markup above is too complex

amivanoff commented:
> Does the validation work with links like :https://github.com/apache/jena/tree/main/jena-shacl ? If so, let's go for that. The third markup above is too complex

Fixed here https://github.com/w3c-cg/awesome-semantic-shapes/blob/22b7da77fb2ec445cd0c0a26bf8465642435b96f/README.md

- No separate [documentation]() links in item's description.
- List item main link is repository subcomponent link (not doc-link).
- Badge <a> links removed for markdown clarity.
- Switched back from markdown emojis to UTF-8 emojis (plain text emoji is not so visible in the editor).
- Added "license;" to the license SPDX codes (not "tags-like" markup in the end of description but more of a "text-like").
- Added Part 2 SHACL 2017 talk by TopQuadrant.

VladimirAlexiev commented on README.md:
```diff
@@ -30,34 +30,34 @@ Semantic shapes are frequently described using the SHACL or ShEx language.
-- [shacl-engine](https://github.com/rdf-ext/shacl-engine) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/rdf-ext/shacl-engine" align="top"> - Supports SHACL Core, SHACL-SPARQL; A fast engine for data provided as [RDF/JS](http://rdf.js.org/data-model-spec/) objects; [playground](https://playground.rdf-ext.org/shacl/); `MIT` `JavaScript`.
-- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/TopQuadrant/shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL, SHACL rules; based on Jena; `BSD-3-Clause` `Java`.
-- [TopBraid SHACL API Extended](https://github.com/SHACL-X/shacl-x) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/SHACL-X/shacl-x" align="top"> - Fork of TopBraid SHACL API + added SHACL-JS based on GraalVM Polyglot; `BSD-3-Clause` `Java`.
+- [Apache Jena SHACL](https://github.com/apache/jena/tree/main/jena-shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/apache/jena" align="top"> - Supports SHACL Core, SHACL-SPARQL; `Apache-2.0` license; `Java`.
```
I thought you'd keep the doc links?!

amivanoff replied:
It should be like this with doc-links?
- [Apache Jena SHACL](https://github.com/apache/jena/tree/main/jena-shacl) <img alt="GitHub Last Release" src="https://img.shields.io/maven-central/v/org.apache.jena/jena-shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL; [docs](https://jena.apache.org/documentation/shacl/index.html); `Apache-2.0` license; `Java`.

amivanoff commented on README.md:
```diff
@@ -1,5 +1,97 @@
+
+- [Jena SHACL](https://github.com/apache/jena/) -- Java, Supports: SHACL Core, SHACL-SPARQL
+- [RDF4J SHACL Sail](https://github.com/eclipse-rdf4j/rdf4j) -- Java
+- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) -- Java, based on Jena, supports: SHACL Core, SHACL-SPARQL, SHACL rules
```
I have doubts about [TopQuadrant SHACL playground](https://shacl.org/playground/):
- It is a JS validator, not Java one. "...a version of the source code can be found at [SHACL-JS GitHub project](https://github.com/TopQuadrant/shacl-js)"
- "Last updated: 2017-05-01 and no longer actively maintained... See also [Zazuko SHACL Playground](https://zazuko.github.io/shacl-playground/) for a newer implementation"

amivanoff replied:
We should change "last commit date" to "last release date" whenever it is possible?

From this (some examples with recent version and old version)
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/TopQuadrant/shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL, SHACL rules; based on Jena; [RDF validator](https://www.itb.ec.europa.eu/shacl/any/upload), [SHACL shapes validator](https://www.itb.ec.europa.eu/shacl/shacl/upload), [Sparna SHACL playground](https://shacl-play.sparna.fr/play/); `BSD-3-Clause` license; `Java`.
- [shexSpec/shex.js](https://github.com/shexjs/shex.js) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/shexjs/shex.js" align="top"> - JavaScript implementation of Shape Expressions; [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html); `MIT` license; `JavaScript`.

To this?
- [TopBraid SHACL API](https://github.com/TopQuadrant/shacl) <img alt="GitHub Last Release" src="https://img.shields.io/github/v/release/TopQuadrant/shacl" align="top"> <img alt="GitHub Release Date" src="https://img.shields.io/github/release-date/TopQuadrant/shacl" align="top"> - Supports SHACL Core, SHACL-SPARQL, SHACL rules; based on Jena; [RDF validator](https://www.itb.ec.europa.eu/shacl/any/upload), [SHACL shapes validator](https://www.itb.ec.europa.eu/shacl/shacl/upload), [Sparna SHACL playground](https://shacl-play.sparna.fr/play/); `BSD-3-Clause` license; `Java`.
- [shexSpec/shex.js](https://github.com/shexjs/shex.js) <img alt="GitHub Last Release" src="https://img.shields.io/github/v/release/shexjs/shex.js" align="top"> <img alt="GitHub Release Date" src="https://img.shields.io/github/release-date/shexjs/shex.js" align="top"> - JavaScript implementation of Shape Expressions; [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html); `MIT` license; `JavaScript`.

bergos commented on README.md:
```diff
@@ -1,5 +1,129 @@
+  - [SHACL Advanced Features 1.1](https://w3c.github.io/shacl/shacl-af/)
+  - [SHACL Compact Syntax](https://w3c.github.io/shacl/shacl-compact-syntax/)
+  - [SHACL JavaScript Extensions](https://w3c.github.io/shacl/shacl-js/)
+
```
Maybe also include other SHACL-related specifications like the [The Shape Topologies algorithm](https://w3id.org/tree/specification/shape-topologies)?

amivanoff replied:
I propose a new subsection like "SHACL-related specifications" with all the random stuff here. Maybe we could also put [DASH Data Shapes](https://datashapes.org) here?

VladimirAlexiev replied:
Sure!

amivanoff replied:
It's in the last commit, already.

bergos commented on README.md:
```diff
@@ -1,5 +1,129 @@
+- [RDF4J SHACL Engine](https://github.com/eclipse-rdf4j/rdf4j/tree/main/core/sail/shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/eclipse-rdf4j/rdf4j" align="top"> - Supports SHACL Core (without some property paths), SHACL-SPARQL, incremental validation; `BSD-3-Clause` license; `Java`.
+- [rdf-validate-shacl](https://github.com/zazuko/rdf-validate-shacl) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/zazuko/rdf-validate-shacl" align="top"> - Supports SHACL Core; pure JavaScript validator on top of the [RDFJS](https://rdf.js.org/) stack; [playground](https://zazuko.github.io/shacl-playground); `MIT` license; `JavaScript`.
+- [SHACL for Ruby](https://github.com/ruby-rdf/shacl) ![GitHub last commit](https://img.shields.io/github/last-commit/ruby-rdf/shacl) - A pure-Ruby library for working with the Shape Constraint Language to validate the shape of RDF graphs; `BSD-3-Clause` license; `Ruby`.
+- [shacl-engine](https://github.com/rdf-ext/shacl-engine) <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/rdf-ext/shacl-engine" align="top"> - Supports SHACL Core, SHACL-SPARQL; A fast engine for data provided as [RDF/JS](http://rdf.js.org/data-model-spec/) objects; [playground](https://playground.rdf-ext.org/shacl/); `MIT` license; `JavaScript`.
```
The command line tool [rdf-ext-cli](https://github.com/rdf-ext/rdf-ext-cli) provides a validation feature based on shacl-engine. Data and shapes can be given as file, URL, or SPARQL endpoint + query.

amivanoff commented on README.md:
```diff
@@ -1,5 +1,97 @@
+## ShEx Validators
+
+- [Apache Jena ShEx](https://jena.apache.org/documentation/shex/index.html) -- Java
+- [shexSpec/shex.js](https://github.com/shexjs/shex.js) -- JS, [playground](http://rawgit.com/shexSpec/shex.js/master/doc/shex-simple.html)
```
I made an example with sub-bullets above

amivanoff replied:
Done

amivanoff commented:
I fixed all the comments from above. Whenever possible, I used shield.io badges for versions and last activity dates because this eliminated the necessity for manual updates.
https://github.com/w3c-cg/awesome-semantic-shapes/blob/09dce6b09215c2ee9a44c075dbcaa0aebe4a3b5c/README.md

VladimirAlexiev commented:
Great job @amivanoff ! I'll merge it but first want to pick up all links mentioned, eg by @bergos.
Then I'll submit a second PR with more tools and goodies.

amivanoff commented:
@VladimirAlexiev, It's added already, under the "rdf-ext" bullet in the "SHACL Validators" section and under the bullet "SHACL-related specifications" in the "Specifications" section.

amivanoff commented on README.md:
```diff
@@ -1,5 +1,97 @@
+  - [Shapes Constraint Language (SHACL)](https://www.w3.org/TR/shacl/) -- W3C Recommendation, 20 July 2017
+  - [SHACL Advanced Features](https://www.w3.org/TR/shacl-af/) -- W3C Working Group Note, 08 June 2017
+  - [SHACL JavaScript Extensions](https://www.w3.org/TR/shacl-js/) -- W3C Working Group Note, 08 June 2017
+  - [SHACL Test Suite and Implementation Report](https://w3c.github.io/data-shapes/data-shapes-test-suite/) -- W3C Document 17 January 2024
```
Fixed
